src/v/raft/consensus.cc

```diff
@@ -2013,7 +2013,7 @@ consensus::do_append_entries(append_entries_request&& r) {
 
     // section 3
     if (request_metadata.prev_log_index < last_log_offset) {
-        if (unlikely(request_metadata.prev_log_index < _commit_index)) {
+        if (unlikely(request_metadata.prev_log_index < last_visible_index())) {
```

Comment:
so IIUC the problem is with follower fetching + acks=1 because, on a follower the log can be truncated below high watermark (because of committed offset check) say during recovery and the subsequent read from the follower may result in an out of range read, is that correct?

Reply:
exactly